### PR TITLE
HBASE-22193 Add backoff when region failed open too many times

### PR DESCRIPTION
**Pull request by infraio**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 254 | master passed |
| +1 | compile | 52 | master passed |
| +1 | checkstyle | 65 | master passed |
| +1 | shadedjars | 257 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 159 | master passed |
| +1 | javadoc | 33 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 259 | the patch passed |
| +1 | compile | 53 | the patch passed |
| +1 | javac | 53 | the patch passed |
| +1 | checkstyle | 65 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 264 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 514 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 169 | the patch passed |
| +1 | javadoc | 32 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 17615 | hbase-server in the patch failed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 19936 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestSCPWithReplicas |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.master.TestSplitWALManager |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleWAL |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.quotas.TestSpaceQuotas |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleAsyncWAL |
|   | hadoop.hbase.util.TestFromClientSide3WoUnsafe |
|   | hadoop.hbase.client.TestConnectionImplementation |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.client.TestAdmin1 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/133 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux e7b96488013a 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 74112e53c9 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/1/testReport/ |
| Max. process+thread count | 5247 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 251 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 264 | master passed |
| +1 | compile | 52 | master passed |
| +1 | checkstyle | 67 | master passed |
| +1 | shadedjars | 256 | branch has no errors when building our shaded downstream artifacts. |
| -1 | findbugs | 155 | hbase-server in master has 11 extant Findbugs warnings. |
| +1 | javadoc | 32 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 236 | the patch passed |
| +1 | compile | 51 | the patch passed |
| +1 | javac | 52 | the patch passed |
| +1 | checkstyle | 67 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 263 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 521 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 193 | the patch passed |
| +1 | javadoc | 32 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 16154 | hbase-server in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 18697 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.master.TestSplitWALManager |
|   | hadoop.hbase.client.TestSplitOrMergeStatus |
|   | hadoop.hbase.quotas.TestSpaceQuotas |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestAdmin1 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/133 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 5b804743a53e 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 494a8efe5f |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/2/artifact/out/branch-findbugs-hbase-server-warnings.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/2/testReport/ |
| Max. process+thread count | 5057 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java [hunk 1]
```diff
@@ -131,6 +131,9 @@
       "hbase.assignment.maximum.attempts";
   private static final int DEFAULT_ASSIGN_MAX_ATTEMPTS = Integer.MAX_VALUE;
 
+  public static final String ASSIGN_NORMAL_ATTEMPTS = "hbase.assignment.normal.attempts";
```
[the remark this answers is not in the document]
Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java]: ASSIGN_RETRY_IMMEDIATELY_MAX_ATTEMPTS?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java [hunk 2]
```diff
@@ -131,6 +131,9 @@
       "hbase.assignment.maximum.attempts";
   private static final int DEFAULT_ASSIGN_MAX_ATTEMPTS = Integer.MAX_VALUE;
 
+  public static final String FAILED_OPEN_ATTEMPTS = "hbase.assignment.failed.open.attempts";
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java]: Srrory for saying again, but maybe name it

ASSIGN_RETRY_IMMEDIATELY_MAX_ATTEMPTS?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 47 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 295 | master passed |
| +1 | compile | 63 | master passed |
| +1 | checkstyle | 68 | master passed |
| +1 | shadedjars | 264 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 194 | master passed |
| +1 | javadoc | 30 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 241 | the patch passed |
| +1 | compile | 51 | the patch passed |
| +1 | javac | 51 | the patch passed |
| +1 | checkstyle | 64 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 250 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 517 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 184 | the patch passed |
| +1 | javadoc | 33 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 17484 | hbase-server in the patch failed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 19888 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.master.TestSplitWALManager |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleWAL |
|   | hadoop.hbase.quotas.TestSpaceQuotas |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleAsyncWAL |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestAdmin1 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/133 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux c4532375a2e8 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / f227eb7aac |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/3/testReport/ |
| Max. process+thread count | 5044 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-133/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**